tests/Unit/Core/Grid/Query/Filter/DoctrineFilterApplicatorTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Core\Grid\Query\Filter;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Grid\Query\Filter\DoctrineFilterApplicator;
+use PrestaShop\PrestaShop\Core\Grid\Query\Filter\SqlFilters;
+
+class DoctrineFilterApplicatorTest extends TestCase
+{
+    /**
+     * @var SqlFilters
+     */
+    private $priceMinMaxFilters;
+
+    /**
+     * @var QueryBuilder
+     */
+    private $queryBuilder;
+
+    /**
+     * @var DoctrineFilterApplicator
+     */
+    private $doctrineFilterApplicator;
+
+    public function setUp()
+    {
+        $this->priceMinMaxFilters = new SqlFilters();
```

Comment:
Instanciating and storing `DoctrineFilterApplicator` into a class field makes sense because it can be re-used in all tests function (although, strictly speaking, some would argue a new object should be created in each tests, but since it's stateless I don't mind personally)

However `priceMinMaxFilters` is only used for your tests related to MIN_MAX filters, although they are the only ones now (because the class was not tested) they should remain independent data and be created in each test function

Reply:
Alright that makes sense. I guess I thought about more how I would write usual class and not how you should write test.

Implemented the changes.